### PR TITLE
#167727310 Endpoints for Social Media Login

### PR DESCRIPTION
**Pull request by stepheng323**
#### What does this PR do?
This PR build the endpoints for social media login. A user can log in to barefoot nomad via facebook or google
#### Description of Task to be completed?

- The following endpoint was set up to handle google and facebook login respectively
 GET api/v1/auth/google
 GET api/vi/auth/facebook
- Setup google login app on Google developer console

- Setup facebook login app on Facebook developer console 

- Setup passport to use both google and facebook strategy

#### How should this be manually tested?

- Run git clone in terminal `https://github.com/andela/nightcrawler-backend.git`
- Cd into the folder created `nightcrawler-backend` and run git fetch `ft--social-media-login-167727310`, finally run npm install
- Make sure your database is seeded with a user that has a valid google/facebook account
- In your .env file add `GOOGLE_CLIENT_ID`, `GOOGLE_CLIENT_SECRET` ,`GOOGLE_CALLBACK` , `FACEBOOK_APP_ID`,  `FACEBOOK_APP_SECRET`, `FACEBOOK_CALLBACK` as shared in the slack channel
- Run npm run start:dev to start the server
- Navigate to `localhost:300/api/v1/auth/google` for google login or `localhost:300/api/v1/auth/facebook`
- Login into google or facebook with an email address registered on barefoot nomad
#### Any background context you want to provide?
- Your database and model needs to be in place.
- enviroment variable as stated above also needs to available
#### What are the relevant pivotal tracker stories?
#167727310

### CODE REVIEW

File: src/config/passport.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import passport from 'passport';
+import GoogleStrategy from 'passport-google-oauth20';
+import FacebookStrategy from 'passport-facebook';
+import * as constant from './constants';
+
+passport.use(new GoogleStrategy({
```
Comment (solomonfrank) [inline src/config/passport.js]: You have done good work bro, but why don't you consider abstracting  the relevant strategies config and then import

File: src/config/passport.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  callbackURL: constant.GOOGLE_CALLBACK,
+  clientID: constant.GOOGLE_CLIENT_ID,
```
Comment (solomonfrank) [inline src/config/passport.js]: You have done good work bro, but why don't you consider  abstracting the relevant strategies and then import the abstracted object into their respective passport.use

**Review by solomonfrank — CHANGES_REQUESTED** [review]

File: src/config/passport.js (hunk 2 above)
Comment (stepheng323) [reply src/config/passport.js]: Ok, I get it. thanks for that.

**Review by joaquinto — COMMENTED** [review]
Looks Good To Me

File: package.json [hunk 3]
```diff
@@ -35,12 +35,17 @@
...
+    "passport-google-oauth20": "^2.0.0",
     "pg": "^7.12.1",
     "pg-hstore": "^2.3.3",
     "sequelize": "^5.16.0",
     "sequelize-cli": "^5.5.0",
+    "sinon": "^7.4.1",
     "swagger-jsdoc": "^3.4.0",
-    "swagger-ui-express": "^4.0.7"
+    "swagger-ui-express": "^4.0.7",
+    "zombie": "^6.1.4"
```
Comment (Hector101) [inline package.json]: uninstall this package if not in use

File: src/index.js [hunk 4]
```diff
@@ -23,6 +25,8 @@ const baseURL = '/api/v1/';
 
 app.use(baseURL, router);
 app.use('/api-docs', swaggerUI.serve, swaggerUI.setup(swaggerDocument));
+app.use(passport.initialize());
```
Comment (Hector101) [inline src/index.js]: we don't need this since we're not using passport sessions.

File: src/modules/auth/auth.controller.js [hunk 5]
```diff
@@ -29,6 +31,27 @@ class AuthController {
...
+* Login a user with social media
+* @param {object} req
+* @param {object} res
+* @returns {object} json response
+*/
+  static async socialAuth(req, res) {
+    const { value } = req.user.emails[0];
+    const currentUser = await User.findOne({ where: { email: value } });
+    if (!currentUser) {
+      return responseHandler.invalidCredential(res);
```
Comment (Hector101) [inline src/modules/auth/auth.controller.js]: Implement a meta redirect on this if statement to redirect to the client-side login page.

**Review by Hector101 — CHANGES_REQUESTED** [review]

**Review by steelze — COMMENTED** [review]
LGTM

File: src/modules/auth/auth.controller.js (hunk 5 above)
Comment (stepheng323) [reply src/modules/auth/auth.controller.js]: done

File: src/index.js (hunk 4 above)
Comment (stepheng323) [reply src/index.js]: done

File: package.json (hunk 3 above)
Comment (stepheng323) [reply package.json]: done

**Pull request merged**